frontend/packages/core/src/stepper.tsx

```diff
@@ -0,0 +1,153 @@
+  padding: "8px",
+}));
+
+const ClearIcon = CheckIcon.withComponent(PriorityHighIcon);
+
+const StepConnector = styled(MuiStepConnector)((props: { completed?: boolean }) => ({
+  ".MuiStepConnector-line": {
+    height: "5px",
+    background: props.completed ? "#3548D4" : "#E7E7EA",
+    border: "0",
+  },
+}));
+
+const StepLabelIcon = styled(MuiStepLabel)({
+  ".MuiStepLabel-iconContainer": {
+    padding: "0",
+  },
+});
+
+type StepIconVariant = "active" | "pending" | "success" | "failed";
+export interface StepIconProps {
+  index: number;
+  variant: StepIconVariant;
+}
+
+const stepIconVariants = {
+  active: {
+    background: "#FFFFFF",
+    border: "1px solid #3548D4",
+    font: "#3548D4",
+  },
+  pending: {
+    background: "#E7E7EA",
+    border: "#E7E7EA",
+    font: "rgba(13, 16, 48, 0.38)",
+  },
+  success: {
+    background: "#3548D4",
+    border: "#3548D4",
+    font: "#FFFFFF",
+  },
+  failed: {
+    background: "#DB3615",
+    border: "#DB3615",
+    font: "#FFFFFF",
+  },
+};
+
+const StepIcon: React.FC<StepIconProps> = ({ index, variant }) => {
+  const color = stepIconVariants[variant || "pending"];
+  let Icon = <>{index}</>;
+  if (variant === "success") {
+    Icon = <CheckIcon font={color.font} fontSize="large" />;
+  } else if (variant === "failed") {
+    Icon = <ClearIcon font={color.font} fontSize="large" />;
+  }
+  return (
+    <Circle background={color.background} border={color.border}>
+      <DefaultIcon font={color.font}>{Icon}</DefaultIcon>
+    </Circle>
+  );
+};
+
+const StepLabel = styled(Grid)(
+  {
+    fontWeight: 500,
+    fontSize: "14px",
+  },
+  props => ({
+    color: props["data-active"] ? "#0D1030" : "rgba(13, 16, 48, 0.38)",
+  })
+);
+
+export interface StepProps {
+  label: string;
+  error?: boolean;
+}
+
+const Step: React.FC<StepProps> = ({ children }) => <>{children}</>;
+
+export interface StepperProps {
+  activeStep: number;
+  children?: React.ReactElement<StepProps>[] | React.ReactElement<StepProps>;
+}
+
+const Stepper = ({ activeStep, children }: StepperProps) => (
+  <div>
+    <MuiStepper style={{ padding: "0" }} activeStep={activeStep + 1} connector={<StepConnector />}>
+      {React.Children.map(children, (step: any, idx: number) => {
+        const stepProps = {
+          index: idx + 1,
+          variant: "pending" as StepIconVariant,
+        };
+        if (idx === activeStep) {
+          stepProps.variant = "active";
+        } else if (idx < activeStep) {
+          stepProps.variant = step.props.error ? "failed" : "success";
+        }
+
+        return (
+          <MuiStep key={step.label} style={{ padding: "0" }}>
+            <StepLabelIcon icon={<StepIcon {...stepProps} />} />
+          </MuiStep>
+        );
+      })}
+    </MuiStepper>
+    <Grid style={{ padding: "16px 0" }} container justify="space-between">
+      {React.Children.map(children, (step: any, idx: number) => (
+        <StepLabel item data-active={idx === activeStep}>
```

Comment:
thinking about it more what about using a class name instead of data-.

Reply:
I kind of like using props here since it makes it clear what we are specifying compared to what Material is injecting

Reply:
though data won't cascade for nested selectors, i think in general we should standardize on classes. we can use a prefix to distinguish from Mui if needed.

note: (comment after merge, not a blocker)

Reply:
👍  going to make a cleanup task and add some of the standardizing bits we have talked about to it